source/gfoidl.Stochastics/Statistics/Sample.AverageVarianceCore.cs

```diff
@@ -20,98 +23,97 @@ private void CalculateAverageAndVarianceCore()
             _varianceCore = variance;
         }
         //---------------------------------------------------------------------
-        internal (double avg, double variance) CalculateAverageAndVarianceCoreSimd()
+        internal void CalculateAverageAndVarianceCoreSimd(out double avg, out double variance)
         {
-            var (avg, variance) = this.CalculateAverageAndVarianceCoreImpl((0, this.Count));
-
-            return (avg, variance);
+            this.CalculateAverageAndVarianceCoreImpl(0, this.Count, out avg, out variance);
         }
         //---------------------------------------------------------------------
-        internal (double avg, double variance) CalculateAverageAndVarianceCoreParallelizedSimd()
+        internal void CalculateAverageAndVarianceCoreParallelizedSimd(out double avg, out double variance)
         {
-            double avg      = 0;
-            double variance = 0;
+            double tmpAvg      = 0;
+            double tmpVariance = 0;
 
             Parallel.ForEach(
                 Partitioner.Create(0, _values.Length),
                 range =>
                 {
-                    var (localAvg, localVariance) = this.CalculateAverageAndVarianceCoreImpl((range.Item1, range.Item2));
-                    localAvg.SafeAdd(ref avg);
-                    localVariance.SafeAdd(ref variance);
+                    this.CalculateAverageAndVarianceCoreImpl(range.Item1, range.Item2, out double localAvg, out double localVariance);
+                    localAvg.SafeAdd(ref tmpAvg);
+                    localVariance.SafeAdd(ref tmpVariance);
                 }
             );
 
-            return (avg, variance);
+            avg      = tmpAvg;
+            variance = tmpVariance;
         }
         //---------------------------------------------------------------------
-        private unsafe (double avg, double variance) CalculateAverageAndVarianceCoreImpl((int start, int end) range)
+        private unsafe void CalculateAverageAndVarianceCoreImpl(int i, int n, out double avg, out double variance)
```

Comment:
`n` was a suboptimal choice for the variable name. Here `n` is the index of the end of the range.
Normally `n` stands for count, etc.

This will be fixed in https://github.com/gfoidl/Stochastics/pull/60 (because I stumbled in that PR about this miss-naming).